### ISSUE TITLE
Enable basic additional operations with setTo

### ISSUE THREAD

**Issue by chrisnurse** [rangav/thunder-client-support#481]
# Description
When executing tests and using setTo it would useful to be able to extract values using basic commands, something that we can do with Postman scripting. However, I don't feel we need an entire scripting capability in order to elevate the power of the setTo option.

We can use a simple expression evaluator to add very valuable features.

# Example use case
After we call a create user API method (for example in keycloak), it returns a Location header:
```http://devhost:8093/auth/admin/realms/app/users/13f239e7-8771-4e75-9c88-341f62650f1b```

# Example syntax
**header.location.split("/")[8]**  // Split the URL into segments and take the approriate element

This would make the value ```13f239e7-8771-4e75-9c88-341f62650f1b``` available as an environment variable.

After creating a test user it is then possible to delete the same user and clean up.

# Postman equivalent
```
let location = pm.response.headers.get('location').split('/');
let user_id = location[8];
postman.setEnvironmentVariable("kc_new_user_id", user_id);
```

**Comment by rangav** [comment]
Thanks @chrisnurse for the feedback, Will add this to roadmap

**Comment by chrisnurse** [comment]
You're very welcome.

Thank you for this fantastic contribution to the community. I am loving Thunder Client!

I can recommend https://github.com/SujiJoe/ts-expression-evaluator 

This has helped me to build several features that need expression evaluation, such as dynamic forms, where some fields need to appear or be invisible based on some flexible logic and values of other input fields. I found it very intuitive.

**Comment by rangav** [comment]
@chrisnurse glad to know you like the extension 🙂

thanks for suggestion, will look into it.

**Comment by namhuynhkien** [comment]
Hi @rangav , 
do you have any update on this or any alternate to achieve the same?

**Comment by rangav** [comment]
@namhuynhkien what is your use-case, can you please share more details

**Comment by namhuynhkien** [comment]
The same as Chris. the response returns the location in the header which contain value needed for next request. e.g. for example I want to get the useCaseId. Below is my script in Postman
`
var loc= pm.response.headers.get('Location');
pm.environment.set("location", loc);
var firstindex = loc.lastIndexOf('=') + 1;
var requestId = loc.substring(firstindex);
pm.environment.set("use_case_id", useCaseId);
`

**Comment by rangav** [comment]
@namhuynhkien thanks for sharing, will implement a solution asap.

**Comment by skela** [comment]
Another useful thing would be to be able to base64 decode for example a jwt, and then use one of the json properties inside to set an env variable.

**Comment by rangav** [comment]
This feature is now implemented and published to marketplace, please update to `v2.0.0`

See all features released
https://github.com/rangav/thunder-client-support/releases/tag/v2.0.0

Please let me know your feedback.

**Comment by rangav** [comment]
When setting the variable can use the format - `header.name | split(" ") | slice(7, 8) | join`

Will implement [at](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at) filter in the next update

**Comment by rangav** [comment]
We have implemented `at` filter in v2.0.1
https://github.com/rangav/thunder-client-support/blob/master/docs/filters.md#at

Now you can `header.location | split("/") | at(8)`

see release notes here
https://github.com/rangav/thunder-client-support/releases/tag/v2.0.1
